pkg/agent/agent.go

```diff
@@ -112,6 +123,41 @@ func (a *Agent) shutdown() {
 	}
 }
 
+func (a *Agent) setupProfiling() {
+	if a.c.ProfilingEnabled {
```

Comment:
This could be easier to read if this conditional was set at the callsite instead, i.e.
```
if a.c.ProfilingEnabled {
     a.setupProfiling()
}
```

Reply:
Agree.